pgmpy/models/NaiveBayesModel.py

```diff
@@ -0,0 +1,224 @@
+        """
+        if not obs_nodes_list:
+            return set()
+        return set(obs_nodes_list) | set(self.parent_node)
+
+
+    def active_trail_nodes(self, start, observed=None):
+        """
+        Returns all the nodes reachable from start via an active trail.
+
+        Parameters
+        ----------
+
+        start: Graph node
+
+        observed : List of nodes (optional)
+            If given the active trail would be computed assuming these nodes to be observed.
+
+        Examples
+        --------
+
+        >>> from pgmpy.models import NaiveBayesModel
+        >>> model = NaiveBayesModel()
+        >>> model.add_nodes_from(['a', 'b', 'c', 'd'])
+        >>> model.add_edges_from([('a', 'b'), ('a', 'c'), ('a', 'd')])
+        >>> model.active_trail_nodes('a')
+        {'a', 'b', 'c', 'd'}
+        >>> model.active_trail_nodes('a', ['b', 'c'])
+        {'a', 'd'}
+        >>> model.active_trail_nodes('b', ['a'])
+        {'b'}
+        """
+
+        if observed and self.parent_node in observed:
+            return set(start)
+        else:
+            return set(self.nodes()) - set(observed if observed else [])
+
+    def local_independencies(self, variables):
+        """
+        Returns a list of independencies objects containing the local independencies
+        of each of the variables. If local independencies does not exist for a variable
+        it gives a None for that variable.
+
+
+        Parameters
+        ----------
+        variables: str or array like
+            variables whose local independencies are to found.
+
+        Examples
+        --------
+        >>> from pgmpy.models import NaiveBayesModel
+        >>> model = NaiveBayesModel()
+        >>> model.add_edges_from([('a', 'b'), ('a', 'c'), ('a', 'd')])
+        >>> ind = model.local_independencies('b')
+        >>> ind
+        [(b _|_ d, c | a)]
+        """
+        independencies = []
+        for variable in [variables] if isinstance(variables, str) else variables:
+            if variable != self.parent_node:
+                independencies.append(Independencies([variable, list(set(self.children_nodes)
+                                                                 - set(variable)), self.parent_node]))
+            else:
+                independencies.append(None)
+        return independencies
+
+    def fit(self, data, parent_node, estimator_type=None):
+        """
+        Computes the CPD for each node from a given data in the form of a pandas dataframe.
+        If a variable from the data is not present in the model, it adds that node into the model. 
+
+        Parameters
+        ----------
+        data : pandas DataFrame object
+            A DataFrame object with column names same as the variable names of network
+
+        parent_node: str
+            parent node of the model
+
+        estimator: Estimator class
+            Any pgmpy estimator. If nothing is specified, the default Maximum Likelihood
```

Comment:
@yashu-seth Replace `Maximum Likelihood estimator` with "`MaximumLikelihoodEstimator`" (notice the backtick). Using backtick will put the name in code snippet in out online docs.